src/worker-helpers.js

```diff
@@ -114,26 +114,14 @@ export function getESLintInstance(fileDir, config, projectPath) {
   return getESLintFromDirectory(modulesDir, config, projectPath)
 }
 
-export function getConfigPath(fileDir) {
-  const configFile =
-    findCached(fileDir, [
-      '.eslintrc.js', '.eslintrc.yaml', '.eslintrc.yml', '.eslintrc.json', '.eslintrc', 'package.json'
-    ])
-  if (configFile) {
-    if (Path.basename(configFile) === 'package.json') {
-      // eslint-disable-next-line import/no-dynamic-require
-      if (require(configFile).eslintConfig) {
-        return configFile
-      }
-      // If we are here, we found a package.json without an eslint config
-      // in a dir without any other eslint config files
-      // (because 'package.json' is last in the call to findCached)
-      // So, keep looking from the parent directory
-      return getConfigPath(Path.resolve(Path.dirname(configFile), '..'))
-    }
-    return configFile
+export function getConfigForFile(eslint, filePath) {
+  const cli = new eslint.CLIEngine()
+  try {
```

Comment:
Is this a dangerous function call that could throw? No use of `try` in api docs.

Reply:
If it can't find a configuration at all it `throw`s an `Error` (hence the comment).